lib/web/apiserver.go

```diff
@@ -605,20 +663,122 @@ func (h *Handler) oidcLoginWeb(w http.ResponseWriter, r *http.Request, p httprou
 	return nil, nil
 }
 
-func (h *Handler) oidcLoginConsole(w http.ResponseWriter, r *http.Request, p httprouter.Params) (interface{}, error) {
-	log.Debugf("oidcLoginConsole start")
-	var req *client.SSOLoginConsoleReq
+func (h *Handler) githubLoginWeb(w http.ResponseWriter, r *http.Request, p httprouter.Params) (interface{}, error) {
+	logger := log.WithFields(log.Fields{trace.Component: "github"})
+	logger.Debug("Web login start")
+	clientRedirectURL := r.URL.Query().Get("redirect_url")
+	if clientRedirectURL == "" {
+		return nil, trace.BadParameter("missing redirect_url query parameter")
+	}
+	connectorID := r.URL.Query().Get("connector_id")
+	if connectorID == "" {
+		return nil, trace.BadParameter("missing connector_id query parameter")
+	}
+	csrfToken, err := csrf.ExtractTokenFromCookie(r)
+	if err != nil {
+		logger.Warnf("Unable to extract CSRF token from cookie: %v", err)
+		return nil, trace.AccessDenied("access denied")
+	}
+	response, err := h.cfg.ProxyClient.CreateGithubAuthRequest(
+		services.GithubAuthRequest{
+			CSRFToken:         csrfToken,
+			ConnectorID:       connectorID,
+			CreateWebSession:  true,
+			ClientRedirectURL: clientRedirectURL,
+			CheckUser:         true,
+		})
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	http.Redirect(w, r, response.RedirectURL, http.StatusFound)
+	return nil, nil
+}
+
+func (h *Handler) githubLoginConsole(w http.ResponseWriter, r *http.Request, p httprouter.Params) (interface{}, error) {
+	log.WithFields(log.Fields{trace.Component: "github"}).Debug(
+		"Console login start")
+	var req client.SSOLoginConsoleReq
 	if err := httplib.ReadJSON(r, &req); err != nil {
 		return nil, trace.Wrap(err)
 	}
-	if req.RedirectURL == "" {
-		return nil, trace.BadParameter("missing RedirectURL")
+	if err := req.Check(); err != nil {
+		return nil, trace.Wrap(err)
+	}
+	response, err := h.cfg.ProxyClient.CreateGithubAuthRequest(
+		services.GithubAuthRequest{
+			ConnectorID:       req.ConnectorID,
+			PublicKey:         req.PublicKey,
+			CertTTL:           req.CertTTL,
+			CheckUser:         true,
```

Comment:
why check user is true here?

Reply:
I've originally implemented it similar to OIDC but then realized that this parameter was always true anyway, so I've just removed it now.